src/tiled/brokenlinks.cpp

```diff
@@ -161,8 +184,19 @@ void BrokenLinksModel::refresh()
                     processTileset(tileset);
                 }
             }
+
+            for (TemplateGroup *templateGroup : mapDocument->map()->templateGroups()) {
+                if (!templateGroup->fileName().isEmpty() && !templateGroup->loaded()) {
+                    BrokenLink link;
+                    link.type = TemplateGroupReference;
+                    link._templateGroup = templateGroup;
+                    mBrokenLinks.append(link);
+                }
+            }
         } else if (auto tilesetDocument = qobject_cast<TilesetDocument*>(mDocument)) {
             processTileset(tilesetDocument->tileset());
+        } else if (auto templateGroupDocument = qobject_cast<TemplateGroupDocument*>(mDocument)) {
+            processTemplateGroup(templateGroupDocument->templateGroup());
```

Comment:
I think you can just leave out this part as well as `processTemplateGroup`. The `mDocument` will never be a `TemplateGroupDocument`, since currently template group documents can't be opened as a stand-alone file. And even when it could, checking the file itself for being loaded makes no sense. The only thing that would make sense then, is to check whether its tileset references and any images referenced by those tilesets have been loaded.

It would be fine to make some preparations, but I think it's a little early. We will get to that point when we actually implement a TemplateGroupEditor.